packages/babel-traverse/src/path/index.ts

```diff
@@ -92,12 +92,57 @@ class NodePath<T extends t.Node = t.Node> {
       // @ts-expect-error key must present in container
       container[key];
 
-    let paths = pathCache.get(parent);
-    if (!paths) {
-      paths = new Map();
-      pathCache.set(parent, paths);
+    if (!IS_STANDALONE) {
+      if (!USE_ESM) {
+        if (!process.env.BABEL_8_BREAKING) {
+          // noHubInCacheKeyForBackwardCompat has three states (true, false, and
+          // undefined) instead of two so that we can run the magic checks only
+          // at the top-level of the traverse call stack, and not when recoursing.
+          // new Error().stack is expensive, and we want to avoid it as much as
+          // possible in this very hot path.
+          // See the .noHubInCacheKeyForBackwardCompat definition in ./cache.ts
+          // for why we need this.
+          // @ts-expect-error ts does not know about this property
+          if (cache.noHubInCacheKeyForBackwardCompat === undefined) {
+            const callerIsOldBabelCore =
```

Comment:
Uhm, this returns `true` even for new babel versions, I have to find an alternative.